src/lib/containers/drug_upload_tool/DrugUploadForm.tsx

```diff
@@ -0,0 +1,733 @@
+          >
+            INCLUDE
+          </a>
+        </div>
+      );
+    } else if (currentStep.excluded === false) {
+      return (
+        <div>
+          This sub-form is currently included in the submission.{' '}
+          <a
+            href="#"
+            onClick={() =>
+              this.showExcludeStateWarningModal(this.state.currentStep.id)
+            }
+          >
+            EXCLUDE
+          </a>
+        </div>
+      );
+    }
+    return <></>;
+  };
+
+  transformErrors = (errors: AjvError[]): AjvError[] => {
+    //if we are not in wizard mode and not rying to submit or validate we just want to skip
+    // over the errors and just set the step status
+    if (
+      this.navAction !== NavActionEnum.SUBMIT &&
+      this.navAction !== NavActionEnum.VALIDATE &&
+      !this.props.isWizardMode
+    ) {
+      const currentStep = {
+        ...this.state.currentStep
+      };
+      if (errors.length > 0) {
+        currentStep.state = StepStateEnum.ERROR;
+      } else {
+        currentStep.state = StepStateEnum.COMPLETED;
+      }
+
+      this.setState({ currentStep });
+
+      return [];
+    }
+
+    // there is an odd behavior in the lib that in cases when we have additional fields depending on enum
+    // value if it's required and not entered we get 3 error: enum, required, and oneOf
+    // so if there is an error Oneof on a parent - ignore it and enum on a child. and just output 'required'
+    // if there is an enum error and there is required with the same prefix remove it
+
+    const reqErrors = errors.filter(error => error.name === 'required');
+    reqErrors.forEach(error => {
+      const parentPath = error.property.substring(
+        0,
+        error.property.lastIndexOf('.')
+      );
+      _.remove(errors, (error: AjvError) => {
+        return (
+          error.property.indexOf(parentPath) > -1 &&
+          (error.name === 'enum' || error.name === 'oneOf')
+        );
+      });
+    });
+    return errors;
+  };
+
+
+
+  renderErrorListTemplate = (props: ErrorListProps) => {
+    let { errors } = props;
+    const currentLis = errors.map((error, i) => {
+      return renderTransformedErrorObject(
+        this.state.steps,
+        error,
+        this.uiSchema,
+        i,
+        this.state.formSchema
+      );
+    });
+
+    return (
+      <div className="form-error-summary">
+        <ul className="error-detail">{currentLis}</ul>
+      </div>
+    );
+  };
+
+  render() {
+    return (
+      <div className="outter-wrap">
+        <Header
+          bodyText={this.state.currentStep.description}
+          title={this.props.formTitle}
+        ></Header>
+        {this.state.isLoading && (
+          <div className="text-center">
+            <span>loading&hellip;</span>
+            <span style={{ marginLeft: '2px' }} className={'spinner'} />
+          </div>
+        )}
+
+        <div
+          style={{
+            display: this.state.isLoading ? 'none' : 'block'
```

Comment:
could remove

Reply:
I think I will wait with this until saving and submission is implemented. Might end up using it. But will put a note